### PR TITLE
Disable the input till the script is loaded.#6357

### PR DESCRIPTION
**Pull request by RohitBhati8269**

### CODE REVIEW

File: web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx [hunk 1]
```diff
@@ -193,6 +193,12 @@ export default function Editor({
             }
             if(update.docChanged) {
               onChange?.(update.view.getValue(), update.view);
+              // Update editable properties
+              editor.current?.dispatch({
```
Comment (adityatoshniwal) [inline web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx:197]: So you're dispatching an event on every doc change. Not at all good for performance.

File: web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx [hunk 2]
```diff
@@ -72,8 +71,15 @@ export default function Query({onTextSelect}) {
   const lastCursorPos = React.useRef();
   const pgAdmin = usePgAdmin();
   const preferencesStore = usePreferences();
-
   const queryToolPref = queryToolCtx.preferences.sqleditor;
+  const [isEditorDisabled, setIsEditorDisabled] = useState(false);
+  const isqueryToolAndUrl = queryToolCtx.params.is_query_tool && queryToolCtx.params.query_url;
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx:76]: this var is not needed.

File: web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx [hunk 3]
```diff
@@ -72,8 +71,15 @@ export default function Query({onTextSelect}) {
...
+
+  useEffect ( () => {
+    if (queryToolCtx.preferences?.sqleditor?.copy_sql_to_query_tool || queryToolCtx.params.is_query_tool && queryToolCtx.params.query_url!== '') {
+      setIsEditorDisabled(true);
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx:80]: where is it enabled again? It should enable again once the script it loaded.

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

File: web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx (hunk 3 above)
Comment (RohitBhati8269) [reply web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx:80]: it enables when the doc updated successfully.

File: web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx (hunk 1 above)
Comment (RohitBhati8269) [reply web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx:197]: As i try to dispatch that event from another component but not worked.
Will check once again.

**Comment by adityatoshniwal** [comment]
What you could do is - create on state variable in qtState - read_only default to true in QueryToolComponent. First thing what query tool does is call `getSQLScript`. If there is no SQL which is going to load then set read_only to false. Otherwise, set it to false in then of API call after EDITOR_SET_SQL.

File: web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx (hunk 1 above)
Comment (RohitBhati8269) [reply web/pgadmin/static/js/components/ReactCodeMirror/components/Editor.jsx]: I have changed the approach as per your suggestions.
Thanks

File: web/pgadmin/tools/sqleditor/static/js/components/QueryToolComponent.jsx [hunk 4]
```diff
@@ -312,27 +312,31 @@ export default function QueryToolComponent({params, pgWindow, pgAdmin, selectedN
       ]
     },
   };
-
-  const getSQLScript = ()=>{
+  
+  const getSQLScript = () => {
+    const setEditorDisabled = (val) => setQtState({ editor_disabled: val });
+    const setSQLAndState = (sql) => {
+      if (sql) eventBus.current.fireEvent(QUERY_TOOL_EVENTS.EDITOR_SET_SQL, sql);
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/QueryToolComponent.jsx:319]: Remove the function setSQLAndState. Call directly.

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

**Pull request merged**